edgelet/iotedged/src/lib.rs

```diff
@@ -557,11 +576,25 @@ where
 
     // Wait for the watchdog to finish, and then send signal to the workload and management services.
     // This way the edgeAgent can finish shutting down all modules.
-    let edge_rt_with_cleanup = edge_rt.map_err(Into::into).and_then(|_| {
-        mgmt_tx.send(()).unwrap_or(());
-        work_tx.send(()).unwrap_or(());
-        future::ok(())
-    });
+
+    let edge_rt_with_cleanup = edge_rt.select2(restart_rx).then(
+        move |res| -> Box<dyn Future<Item = _, Error = _> + Send + Sync> {
```

Comment:
You may be able to get away with not specifying a return type for this closure. Also, you may not need to `Box` the future.